src/spinner/src/Spinner.js

```diff
@@ -38,84 +38,63 @@ const innerClass = color =>
     fill: 'transparent'
   }).toString()
 
-class Spinner extends PureComponent {
-  static propTypes = {
-    /**
-     * Composes the Box component as the base.
-     */
-    ...Box.propTypes,
+function Spinner({ delay = 0, size = 40, ...props }, ref) {
+  const theme = useTheme()
+  const [isVisible, setIsVisible] = useState(delay === 0)
 
-    /**
-     * Delay after which spinner should be visible.
-     */
-    delay: PropTypes.number,
-
-    /**
-     * The size of the spinner.
-     */
-    size: PropTypes.number.isRequired,
-
-    /**
-     * Theme provided by ThemeProvider.
-     */
-    theme: PropTypes.object.isRequired
-  }
-
-  static defaultProps = {
-    size: 40,
-    delay: 0
-  }
-
-  constructor({ delay }) {
-    super()
-
-    this.state = {
-      isVisible: delay === 0
+  useEffect(() => {
+    let delayTimer = null
+    if (delay > 0) {
+      delayTimer = setTimeout(() => {
+        setIsVisible(true)
+      }, delay)
     }
-  }
 
-  render() {
-    if (!this.state.isVisible) {
-      return null
+    return function() {
+      clearTimeout(delayTimer)
     }
+  }, [])
```

Comment:
I got 2 small suggestions:
- I believe we should rerun the effect when the delay prop changes (I know it will only have an effect before the spinner is visible, so maybe it's overkill but I think the eslint-plugin-react-hooks would have warned for this, so we should add it)
- you can return  the cleanup function inside the if statement (less and more obvious code)

```javascript
useEffect(() => {
  if (delay > 0) {
    let timeout = setTimeout(() => {
      setIsVisible(true)
    }, delay)

    return () => clearTimeout(timeout)
  }
}, [delay])
```

Reply:
I think the current implementation mimics what we have today (only `componentDidMount` actually does anything with `delay` – changing it to another value will not change the behavior.

Reply:
As you wish, but don't you think we should add the [eslint-plugin-react-hooks](https://www.npmjs.com/package/eslint-plugin-react-hooks) to at least warn when something like this (see: react-hooks/exhaustive-deps)  happens?